src/CommunityToolkit.Maui/Behaviors/ICommunityToolkitBehavior.shared.cs

```diff
@@ -21,11 +21,40 @@ internal bool TrySetBindingContextToAttachedViewBindingContext()
 			throw new InvalidOperationException($"{nameof(ICommunityToolkitBehavior<TView>)} can only be used for a {nameof(Behavior)}");
 		}
 
-		if (behavior.IsSet(BindableObject.BindingContextProperty) || View is null)
+		if (View is null)
 		{
 			return false;
 		}
 
+		var parent = View.Parent;
+		var assignBindingContext = behavior.IsSet(BindableObject.BindingContextProperty) is false;
+		
+		// If we have a BindingContext, the type is the same as the Views BindingContext and we are inside a CollectionView
+		// then we can assume that we are recycling views and need to assign the BindingContext again.
+		if (View.BindingContext?.GetType() == behavior.BindingContext?.GetType())
+		{
+			for (var i = 0; i < 10; i++)
+			{
+				if (parent is null)
+				{
+					break;
+				}
+
+				if (parent is CollectionView)
```

Comment:
I believe for testing this is enough for now... But collectionView isn't the only control that may have virtualization

Reply:
Good point!

Reply:
still not enough, because devs can apply virtualization to any kind of control or layout... I'm still didn't get the reason to be so conservative on allowing changing the BindingContext. In other words, we shouldn't prevent the BindingContext to be mutable. You may have cases where the dev reuses the control across the page and wants the control's BindingContext to change when that happens

Reply:
I see that we want to automatically inherit the `BindingContext` from the `View` it is attached to **only if the `BindingContext`** is not already set. It appears there are issues around checking whether the `BindingContext` is already set especially when the behavior is used inside something like the `CollectionView`  and the behavior is attached to recycled views.

Reply:
Yeah, but the thing is that after the first set, that value will always be `true`, and that's causing the issue. Because of that I removed the `IsSet` check, as mentioned on the PR description.

I think this happens because when setting the new value, they don't use the [RemoveBinding](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.bindableobject.removebinding?view=net-maui-8.0) method, just `BindinContext = someValue`, causing the `IsSet` to never be false again

Reply:
I didn't confirm the instances but I did discover that OnAttachedTo was being called before OnDetachedFrom when the recycling was happening. Which may explain why IsSet is always true after the first time.